### ISSUE TITLE
YAML.pm doesn't roundtrip with itself [rt.cpan.org #69450]

### ISSUE THREAD

**Issue by ingydotnet** [ingydotnet/yaml-pm#115]
https://rt.cpan.org/Ticket/Display.html?id=69450

```
tom@blip ~ $ perl -MYAML -e 'Load(Dump({ foo => join "", q{"}x40000 }))'
Complex regular subexpression recursion limit (32766) exceeded at
/opt/perlbrew/perls/perl-5.12.3/lib/site_perl/5.12.3/YAML/Loader.pm line
535.
YAML Error: Can't parse single quoted string
   Code: YAML_PARSE_ERR_BAD_SINGLE
   Line: 2
   Document: 1
 at /opt/perlbrew/perls/perl-5.12.3/lib/site_perl/5.12.3/YAML.pm line 36


This is obviously a contrived example, but we've run into real world
cases of it breaking.
```

**Comment by perlpunk** [comment]
This should be fixed in master by now, version 1.25 will be out soon

**Comment by perlpunk** [comment]
Released YAML 1.24_001
